expression/aggregation/base_func.go

```diff
@@ -443,5 +444,31 @@ func (a *baseFuncDesc) WrapCastForAggArgs(ctx sessionctx.Context) {
 		originTp := a.Args[i].GetType().Tp
 		*(a.Args[i].GetType()) = *(a.RetTp)
 		a.Args[i].GetType().Tp = originTp
+		// refine each mysql integer type to the needed decimal precision for sum
```

Comment:
Besides `minimalDecimalLenForHoldingInteger`, how about extract L447-L454 to another function? `adjustDecimalLenForSumInteger`?

Reply:
This function is so tightly coupled with the surrounding code (like the iterating index `i`, and the if-true-then-modify logic), which results in many input/output of the extracted function (not so clean abstraction).

Reply:
```go
func adjustDecimalLenForSumInteger(ft *FieldType, tpOld byte) {
    if types.IsTypeInteger(tpOld) && ft.Tp == mysql.TypeNewDecimal {
        if flen, err := minimalDecimalLenForHoldingInteger(tpOld); err == nil {
            ft.Flen = mathutil.Min(ft.Flen, flen+ft.Decimal)
        }
    }
}
```

Reply:
then here is:
```go
if a.Name == ast.AggFuncSum {
    adjustDecimalLenForSumInteger(a.Args[i].GetType(), tpOld)
}
```

An explicit function can be more readable for readers who don't care the details.